### ISSUE TITLE
Worldhopper crashes and requires a runelite restart on bad connections

### ISSUE THREAD

**Issue by paulocsanz** [runelite/runelite#11819]
**Describe the bug**
The worldhopper seems to be crashing a lot here (my connection is not that good), so the plugin tab keeps blank. And I can't use it from plugins if it has crashed.

**To Reproduce**
1. Have a jittery connection
2. Open runelite
3. Wait
4. See stacktrace

**Expected behavior**
The worldhopper should recover from a bad connection, restarting it again.

**Screenshots**
Sadly I don't have a screenshot right now, if it happens again I will update this issue

**Stacktace**
You can find the stacktrace here: https://pastebin.com/MrWJ5KwK

**Environment (please complete the following information):**
 - OS: Ubuntu
 - RuneLite version: 1.6.19 (i've tracked this bug since 1.6.17)
 - Launcher version: Compiled myself

**Comment by paulocsanz** [comment]
You can consistently see the stacktrace if you kill your connection after the plugin has started. But if it happens after the worldsupplier has filled the plugin tab it doesn't stay blank

**Comment by Adam-** [comment]
I don't think this requires a restart, it would just have to wait until the next world load attempt, which is every 30 minutes.

**Comment by UgiR** [comment]
10 minutes, right?

https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/game/WorldService.java#L52

WorldHopper subscribes to 'world fetch' updates. Or am I misunderstanding.. new here :P

**Comment by Adam-** [comment]
> 
> 
> 10 minutes, right?
> 
> https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/game/WorldService.java#L52
> 
> WorldHopper subscribes to 'world fetch' updates. Or am I misunderstanding.. new here :P

Oh yes, 10 minutes

**Comment by paulocsanz** [comment]
Maybe we could have something detect it failed and retry earlier? There are triggers when connection comes back, why not use it?

If the client's connection is on and it can't connect then it's a worldhopper service problem and it can depend on longer retries (exponential backoff ideally).
